packages/url_launcher/url_launcher_web/lib/link.dart

```diff
@@ -0,0 +1,295 @@
+      <int, _LinkViewController>{};
+
+  static html.Element _viewFactory(int viewId) {
+    return _instances[viewId]?._element;
+  }
+
+  static int _hitTestedViewId;
+  static _ClickListener _hitTestedClickCallback;
+
+  static StreamSubscription _clickSubscribtion;
+
+  static void _onGlobalClick(html.MouseEvent event) {
+    final int viewId = _getViewIdFromTarget(event);
+    _instances[viewId]?._onDomClick(event);
+    // After the DOM click event has been received, clean up the hit test state
+    // so we can start fresh on the next click.
+    _unregisterHitTest();
+  }
+
+  /// Call this method to indicated that a hit test has been registered for the
+  /// given [controller].
+  ///
+  /// The [onClick] callback is invoked when the anchor element receives a
+  /// `click` from the browser.
+  static void _registerHitTest(
+    _LinkViewController controller, {
+    @required _ClickListener onClick,
+  }) {
+    _hitTestedViewId = controller.viewId;
+    _hitTestedClickCallback = onClick;
+  }
+
+  static void _unregisterHitTest() {
+    _hitTestedViewId = null;
+    _hitTestedClickCallback = null;
+  }
+
+  @override
+  final int viewId;
+
+  html.Element _element;
+  bool get _isInitialized => _element != null;
+
+  Future<void> _initialize() async {
+    _element = html.Element.tag('a');
+    setProperty(_element, _viewIdProperty, viewId);
+    _element.style
+      ..opacity = '0'
+      ..display = 'block'
+      ..cursor = 'unset'
+      ;
+
+    final Map<String, dynamic> args = <String, dynamic>{
+      'id': viewId,
+      'viewType': viewType,
+    };
+    await SystemChannels.platform_views.invokeMethod<void>('create', args);
+  }
+
+  void _onDomClick(html.MouseEvent event) {
+    final bool isHitTested = _hitTestedViewId == viewId;
+    if (isHitTested) {
+      _hitTestedClickCallback(event);
+    } else {
+      // There was no hit test registered for this click. This means the click
+      // landed on the anchor element but not on the underlying widget. In this
+      // case, we prevent the browser from following the click.
+      event.preventDefault();
+    }
+  }
+
+  void _setUri(Uri uri) {
+    assert(_isInitialized);
+    if (uri == null) {
+      _element.removeAttribute('href');
+    } else {
+      _element.setAttribute('href', uri.toString());
+    }
+  }
+
+  void _setTarget(LinkTarget target) {
+    assert(_isInitialized);
+    _element.setAttribute('target', _getHtmlTarget(target));
+  }
+
+  String _getHtmlTarget(LinkTarget target) {
+    switch (target) {
+      case LinkTarget.defaultTarget:
+      case LinkTarget.self:
+        return '_self';
+      case LinkTarget.blank:
+        return '_blank';
+      default:
+        throw Exception('Unknown LinkTarget value $target.');
+    }
+  }
+
+  @override
+  Future<void> clearFocus() async {
+    // Currently this does nothing on Flutter Web.
+    // TODO(het): Implement this. See https://github.com/flutter/flutter/issues/39496
```

Comment:
This was copied from the [similar implementation](https://github.com/flutter/flutter/blob/580a96f9576a6026ab74eb92258812d8b3010119/packages/flutter/lib/src/widgets/platform_view.dart#L406-L409) in the framework. Not sure what's the right thing to do here @hterkelsen.